src/object_store.cpp

```diff
@@ -517,6 +517,19 @@ void ObjectStore::verify_compatible_for_read_only(std::vector<SchemaChange> cons
     verify_no_errors<InvalidSchemaChangeException>(verifier, changes);
 }
 
+void ObjectStore::verify_compatible_for_read_only_alternative(std::vector<SchemaChange> const& changes)
+{
+    using namespace schema_change;
+    struct Verifier : SchemaDifferenceExplainer {
+        using SchemaDifferenceExplainer::operator();
+
+        void operator()(RemoveProperty) { }
```

Comment:
> What's the argument for this behaving differently than the immutable mode? It's not clear to me why this mode should throw if the app's schema contains classes not present in the Realm when immutable mode doesn't.

No argument for that actually. It is simply because of java wants to keep current behaviour to throw earlier. 

> Sure, that's roughly how Cocoa implements its equivalent of "wait for initial remote data" as well. But what about using this schema mode without having to wait for all remote changes to be downloaded? Requiring that the set of classes in the schema match prevents using it for this. See realm/realm-cocoa#5186 for a scenario where this would be useful.

Yeah, that is the valid case that the schema validation should not throw. @cmelchior Any comments?